crates/sui/src/keytool.rs

```diff
@@ -37,9 +42,22 @@ impl KeyToolCommand {
     pub fn execute(self, keystore: SuiKeystore) -> Result<(), anyhow::Error> {
         match self {
             KeyToolCommand::Generate => {
-                let (address, keypair) = get_key_pair();
-                store_and_print_keypair(address, keypair)
+                let (_address, keypair) = get_key_pair();
+
+                let hex = encode_bytes_hex(keypair.public_key_bytes());
+                let file_name = format!("{hex:.8}.key");
```

Comment:
Nit: The pubkey is 16 nibbles, which I think is an OK file name length.

Reply:
i'll change that then